updated ROCm tools

smuzaffar opened the pull request:
- No need to define `LIBDIR` for `rocm-rocrand.xml` as it depends on `rocm`
- enable symlink creation for `rocm` libs. Previously this was disabled to avoid getting `rocm/llvm` libs which conflicits with llvm libs from cmssw externals
- Fix `ROOT_INCLUDE_PATH` to have all `INCLUDE` directories. If a tool have multiple include paths then `join="1"` attribute is needed

FYI @fwyzard

cmsbuild commented:
A new Pull Request was created by  @smuzaffar (Malik Shahzad Muzaffar) for branch IB/CMSSW_12_4_X/master.

@cmsbuild, @smuzaffar, @aandvalenzuela, @iarspider can you please review it and eventually sign? Thanks.
@perrotta, @dpiparo, @qliphy you are the release manager for this.
cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

smuzaffar commented:
please test with https://github.com/cms-sw/cms-bot/pull/1751

fwyzard commented:
> * If a tool have multiple include paths then `join="1"` attribute is needed

Is this a new feature ?
At least, I didn't know about it.

smuzaffar commented:
yes, this is recently add. I was also not aware of this issue :-) for the following like, scram uses the last value of `$INCLUDE` and set it to `ROOT_INCLUDE_PATH`
```
<runtime name="ROOT_INCLUDE_PATH" value="$INCLUDE" type="path"/>
```
I added `join` attribute to add all the include paths of the tool

fwyzard commented:
what about the case where `ROOT_INCLUDE_PATH` is given more than once, like [scram-tools.file/tools/vecgeom/vecgeom_interface.xml](https://github.com/cms-sw/cmsdist/blob/IB/CMSSW_12_4_X/master/scram-tools.file/tools/vecgeom/vecgeom_interface.xml) ?

```xml

<tool name="vecgeom_interface" version="@TOOL_VERSION@">
  <info url="https://gitlab.cern.ch/VecGeom/VecGeom"/>
  <client>
    <environment name="VECGEOM_INTERFACE_BASE" default="@TOOL_ROOT@"/>
    <environment name="INCLUDE" default="$VECGEOM_INTERFACE_BASE/include"/>
    <environment name="INCLUDE" default="$VECGEOM_INTERFACE_BASE/include/VecGeom"/>
  </client>
  <runtime name="ROOT_INCLUDE_PATH" value="$VECGEOM_INTERFACE_BASE/include" type="path"/>
  <runtime name="ROOT_INCLUDE_PATH" value="$VECGEOM_INTERFACE_BASE/include/VecGeom" type="path"/>
  <use name="root_cxxdefaults"/>
</tool>
```

smuzaffar commented:
> > I added `join` attribute to add all the include paths of the tool
> 
> By the way, why did you decide to add a new attribute, instead of directly implementing the new behaviour ? Is it to preserve the old behaviour in the other files ?
> 
> From a quick `grep`, it looks like th eonly other tool file with multiple `INCLUDE` lines feeding into a `ROOT_INCLUDE_PATH` is [scram-tools.file/tools/xrootd/xrootd.xml](https://github.com/cms-sw/cmsdist/blob/IB/CMSSW_12_4_X/master/scram-tools.file/tools/xrootd/xrootd.xml) .

I added the new attribute so that it does not break any existing tool definition. Now one can use the new `join` attribute 

```
<runtime name="ROOT_INCLUDE_PATH" value="$INCLUDE" type="path" join="1"/>
```
OR explicit multiple entries ( just like the one in vecgeom_interface.xml)
```
  <runtime name="ROOT_INCLUDE_PATH" value="$VECGEOM_INTERFACE_BASE/include" type="path"/>
  <runtime name="ROOT_INCLUDE_PATH" value="$VECGEOM_INTERFACE_BASE/include/VecGeom" type="path"/>
```

both are same.

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8d5fd1/24135/summary.html
**COMMIT**: 4f053880135ef5fa1697a4437c4c4cf26f79e60c
**CMSSW**: CMSSW_12_4_X_2022-04-22-1100/slc7_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmsdist/7815/24135/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-8d5fd1/39434.75_TTbar_14TeV+2026D88_HLT75e33+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HLT75e33+HARVESTGlobal

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 49
 - DQMHistoTests: Total histograms compared: 3695434
 - DQMHistoTests: Total failures: 8
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3695404
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 48 files compared)
 - Checked 205 log files, 45 edm output root files, 49 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_4_X_2022-04-22-1100+8d5fd1/49788/triggerResults): no differences found
